dcos/client.go

```diff
@@ -1,16 +1,46 @@
 package dcos
 
-import "net/http"
+import (
+	"fmt"
+	"net/http"
+)
 
+// Client
 type Client struct {
-	HttpClient *http.Client
+	HTTPClient *http.Client
+	Config     *Config
+	UserAgent  string
+}
+
+// NewClient returns a Client which will detect its Config through the well
+// known process and use a default http.Client with DC/OS authentication.
+func NewClient() (*Client, error) {
+	config, err := NewConfigManager(nil).Current()
+	if err != nil {
+		return nil, err
+	}
+
+	httpClient, err := NewHTTPClient(config)
+	if err != nil {
+		return nil, err
+	}
 
-	baseURL string
+	return NewClientWithOptions(httpClient, config)
 }
 
-func NewClient(httpClient *http.Client) (*Client, error) {
+// NewClientWithOptions creates a Client from a given *http.Client and *Config
+func NewClientWithOptions(httpClient *http.Client, config *Config) (*Client, error) {
```

Comment:
In the future we should consider using more fine-grained options here (baseURL, acstoken, etc.), as consumers might want to create a client without having a DC/OS CLI config.